src/lib.rs

```diff
@@ -136,7 +137,7 @@ pub extern "C" fn dalek_ed25519_sign(m: *const u8, mlen: UInt, sk: *mut u8, pk:
 
 /// Batch verify signatures, valid[i] == 1 for valid, 0 otherwise
 // TODO(@ryankurte): `ed25519-donna-batchverify.h` has -a lot- going on, presumably for performance reasons (see `cargo bench`)...
-// seems like [`ed25519_dalek::verify_batch`] could substitute but we still need to return the *valid values per message
+// seems like [`ed25519_dalek::verify_batch`] could substitute but we still need to return the *valid values per message (and run without `std` or `alloc`)
 // TODO(@ryankurte): reverse engineer the error returns from the existing code
 #[no_mangle]
 pub extern "C" fn dalek_ed25519_sign_open_batch(m: *mut *const u8, mlen: *mut UInt, pk: *mut *const u8, rs: *mut *const u8, num: UInt, valid: *mut c_int) -> c_int {
```

Comment:
@isislovecruft [ed24419_dalek](https://github.com/dalek-cryptography/ed25519-dalek/blob/main/src/batch.rs#L221) has a handy batch verify function, however, it requires `std` (or `alloc`) and does not fill the array of flags needed by the `donna` API.

for now i have implemented a naive loop but this is substantially less performant than the `donna` version (see `cargo bench`):
```
test bench_batch_verify_dalek ... bench:   5,249,397 ns/iter (+/- 92,808)
test bench_batch_verify_donna ... bench:   1,033,521 ns/iter (+/- 28,110)
```

are you able to look at implementing a compatible non-allocating method either here or in the root crate?